src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs

```diff
@@ -281,12 +283,29 @@ private async Task RegisterVisualStudioTokenCacheAsync(IPublicClientApplication
                 var helper = await MsalCacheHelper.CreateAsync(storageProps);
                 helper.RegisterCache(app.UserTokenCache);
 
-                Context.Trace.WriteLine("Visual Studio token cache configured.");
+                Context.Trace.WriteLine("Microsoft developer tools token cache configured.");
+            }
+            else if (PlatformUtils.IsMacOS())
+            {
+                Context.Trace.WriteLine("Using token cache shared with Microsoft developer tools on macOS...");
+
+                // The shared MSAL cache is located at "~/.local/.IdentityService/msal.cache" on UNIX.
+                string cacheDirectory = Path.Combine(Context.FileSystem.UserHomePath, ".local");
+
+                var storageProps = new StorageCreationPropertiesBuilder(
+                        cacheFileName, cacheDirectory, app.AppConfig.ClientId)
+                    .WithMacKeyChain("Microsoft.Developer.IdentityService", "MSALCache")
+                    .Build();
+
+                var helper = await MsalCacheHelper.CreateAsync(storageProps);
+                helper.RegisterCache(app.UserTokenCache);
+
+                Context.Trace.WriteLine("Microsoft developer tools token cache configured.");
             }
             else
             {
                 string osType = PlatformUtils.GetPlatformInformation().OperatingSystemType;
-                Context.Trace.WriteLine($"Visual Studio token cache integration is not supported on {osType}.");
+                Context.Trace.WriteLine($"Token cache integration is not supported on {osType}.");
```

Comment:
I suppose you could try KeyRing first...

Reply:
Done!